hs_core/views/resource_folder_hierarchy.py

```diff
@@ -138,23 +139,29 @@ def data_store_structure(request):
     for index, fname in enumerate(store[1]):  # files
         f_store_path = os.path.join(store_path, fname)
         file_in_irods = resource.get_irods_path(f_store_path)
-        size = store[2][index]
-        mtype = get_file_mime_type(fname)
-        idx = mtype.find('/')
-        if idx >= 0:
-            mtype = mtype[idx + 1:]
 
+        res_file = None
         if is_federated:
-            f = ResourceFile.objects.filter(object_id=resource.id,
```

Comment:
For each file this generates 1 db query. If a directory has thousands of files, it can significantly delay in generating the response for the UI to display the list of files.